ui/src/picker.test.tsx

```diff
@@ -159,6 +159,19 @@ describe('Picker.tsx', () => {
     expect(queryByRole('option')?.querySelector('.ms-Suggestions-none')).toBeInTheDocument()
   })
 
+  it('Filters correctly - does not offer already selected - without input provided', () => {
+    const { getByRole, queryByRole } = render(<XPicker model={pickerProps} />)
+    const input = (getByRole('combobox') as HTMLInputElement)
+
+    typeToInput(input, name)
+    fireEvent.click(getByRole('option'))
+    typeToInput(input, altName)
+    fireEvent.click(getByRole('option'))
+    fireEvent.click(input)
```

Comment:
Let's also check if we get the expected options step by step, e.g. render 2 choices, select 1, 1 should be present, select the last one, none should be present.